src/Extensions/SearchServiceExtension.php

```diff
@@ -74,8 +75,7 @@ public function updateCMSFields(FieldList $fields): void
             _t(self::class . '.LastIndexed', 'Last indexed in search')
         );
 
-        if ($fields->hasTabSet()) {
-            $fields->addFieldToTab('Root.Main', $showInSearchField);
+        if ($this->owner instanceof SiteTree) {
```

Comment:
Did you consider doing a check for a field of the same name too? eg if there is already a field don't add it.

Reply:
As the field already exists and it's added in `getCMSFields` to another tab, so finding the field is not an option here. Also if I remove this then for `DataObject` the Show in Search field is show at top of all it's fields which does not look good in ui perspective, hence this check is added.